org.eclipse.jface.text/src/org/eclipse/jface/text/source/SourceViewer.java

```diff
@@ -1320,4 +1316,23 @@ public void setCodeMiningAnnotationPainter(AnnotationPainter painter) {
 		ensureCodeMiningManagerInstalled();
 	}
 
+	@Override
+	public void addTextViewerLifecycle(ITextViewerLifecycle lifecycle) {
```

Comment:
I don't think this have to be public at the moment. The viewer is responsible for calling install/uninstall on its "contributors", so this should have to be open.

Reply:
Good catch! Now the method is private and ISourceViewerExtension6 has been removed.